### PR TITLE
Zb/update queued request controller

### PR DESCRIPTION
**Pull request by BelfordZ**
## **Description**

This PR updates the queued request controller to version 0.8.0 - which includes fixes for request queue flushing. See [here for more details](https://github.com/MetaMask/core/blob/main/packages/queued-request-controller/CHANGELOG.md#080)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24061?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

Scenario 1:
1. from dapp 1, call switchEthereumChain, minimize the confirmation
1. from dapp 2, call sendTransaction, minimize the confirmation
1. from dapp 1, call sendTransaction
1. accept the first switchEthereumChain dialog
1. reject the sendTransaction for dapp 2
1. there should be no remaining confirmations.

Scenario 2:
1. from a dapp, call sendTransaction, minimize the confirmation
1. from the same dapp, call wallet_revokePermissions
1. the confirmation should be cleared (i.e. you should not have any pending confirmations)

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -149,6 +149,10 @@ import {
   getLocalizedSnapManifest,
   stripSnapPrefix,
 } from '@metamask/snaps-utils';
+import {
+  methodsRequiringNetworkSwitch,
+  methodsWithConfirmation,
+} from '../../shared/constants/methods-tags';
 ///: END:ONLY_INCLUDE_IF
```
Comment (adonesky1) [inline app/scripts/metamask-controller.js:156]: seems like this shouldn't be in the `ONLY_INCLUDE_IF(keyring-snaps)` block

Comment (BelfordZ) [reply app/scripts/metamask-controller.js:156]: thanks! I was sure I pasted it else where.. guess not woops!

Comment (BelfordZ) [reply app/scripts/metamask-controller.js]: errrr yeah so I guess I was correct to be sure... something wrong with our lint:fix:
https://github.com/MetaMask/metamask-extension/pull/24061/commits/20f03221195b4f42c348f4717d862899fed6c51d

Comment (BelfordZ) [reply app/scripts/metamask-controller.js]: lol more like `yarn lint:break` am i right?

**Comment by BelfordZ** [comment]
@metamaskbot update-policies

**Comment by adonesky1** [comment]
So I tested scenario 1 and only seeing one thing that's a bit weird. Maybe not something we need to solve but just to note, after rejecting the switch call from dapp 1 we don't see the confirmations for dapp 2: they are unapproved in the activity list but don't pop up. If you click out and then reclick into the wallet they will pop up.

https://github.com/MetaMask/metamask-extension/assets/34557516/dfd09591-596b-45d4-a659-db7ef60c8b76

**Comment by adonesky1** [comment]
Also both of the manual test cases here would be good E2E tests to add cc @tmashuang

**Comment by adonesky1** [comment]
Another good one to add to tests - did it manually here with slightly unexpected but not obviously broken UX:

- from dapp1, call sendTransaction
- from dapp2,  call sendTransaction,
- from dapp1 call wallet_revokePermissions
- You should now only see 1 transaction pending (from dapp2)


https://github.com/MetaMask/metamask-extension/assets/34557516/ce03031c-e664-4421-9f37-446bef2760a8

**Comment by adonesky1** [comment]
@SocketSecurity ignore npm/@metamask/queued-request-controller@0.8.0

**Comment by BelfordZ** [comment]
> So I tested scenario 1 and only seeing one thing that's a bit weird. Maybe not something we need to solve but just to note, after rejecting the switch call from dapp 1 we don't see the confirmations for dapp 2: they are unapproved in the activity list but don't pop up. If you click out and then reclick into the wallet they will pop up.
> 

Nice find. The code responsible for popping up the extension when there is confirmations is generally this: https://github.com/MetaMask/metamask-extension/blob/915aa5b8f0164592a587b2b908fa3a255983a9bc/app/scripts/background.js#L844

Im thinking theres prolly a small change to make there to make it work as expected. Do you think we should make a new issue for this one, or get it resolved in this PR?

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by adonesky1 — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**